pyproject.toml

```diff
@@ -71,49 +71,161 @@ line-length = 79
+  # We currently don't conform but we might want to reconsider
+  "G001", # logging-string-format
+  # We currently don't conform but we might want to reconsider
+  "G004", # logging-f-string
+  "I001", # unsorted-imports
+  "N802", # invalid-function-name
+  "N803", # invalid-argument-name
+  "N806", # non-lowercase-variable-in-function
+  # Consider conforming with NPY002
+  "NPY002", # numpy-legacy-random
+  "PLR0904", # too-many-public-methods
+  "PLR0912", # too-many-branches
   "PLR0913", # too-many-arguments
+  "PLR0914", # too-many-locals
+  "PLR0915", # too-many-statements
+  "PLR1702", # too-many-nested-blocks
+  "PLR1704", # redefined-argument-from-local
+  "PLR2004", # magic-value-comparison
+  "PLW1514", # unspecified-encoding
+  "PLW2901", # redefined-loop-name
+  "Q000", # bad-quotes-inline-string
+  "Q004", # unnecessary-escaped-quote
   "UP009", # utf8-encoding-declaration
+  "UP015", # redundant-open-modes
   "UP032", # f-string
-  "UP038" # non-pep604-isinstance
+#  "UP038" # non-pep604-isinstance
   ]
+# Ignored in pylint setup but missing on ruff. We shall delete from the
+# following lines what is not intended to follow anymore.
+# arguments-renamed
+# consider-using-f-string
+# raw-checker-failed
+# bad-inline-option
+# locally-disabled
+# file-ignored
+# suppressed-message
+# useless-suppression
+# deprecated-pragma
+# protected-access
+# redefined-outer-name
+# redefined-builtin
+# broad-except
+# logging-format-interpolation
+# logging-fstring-interpolation
+# wrong-import-order
+# wrong-import-position
+# relative-beyond-top-level
+# too-many-instance-attributes
+# too-few-public-methods
+# invalid-name
+# import-error
+# try-except-raise
+# no-else-raise
+# no-else-return
+# unexpected-keyword-arg
+# no-value-for-parameter
+# too-many-lines
+# arguments-differ
+# import-outside-toplevel
+# super-init-not-called
+# isinstance-second-argument-not-valid-type
+# inconsistent-return-statements
+# no-else-break
+# too-many-function-args
+# redundant-keyword-arg
+# c-extension-no-member
+
+
+
+
+
 
 [tool.ruff.lint.per-file-ignores]
 "__init__.py" = [
   "F401", # unused-import
   ]
-"docs/source/conf.py" = [
-  "E402", # unused-import
-  ]
+#"docs/source/conf.py" = [
+#  "E402", # unused-import
+#  ]
+
+[tool.ruff.lint.pylint]
+max-args = 5 # (PLR0913) Maximum number of arguments for function / method
+max-bool-expr = 5 # ( PLR0916) Boolean in a single if statement
+max-branches=12 # (PLR0912) branches allowed for a function or method body
+max-locals=15 # (PLR0912) local variables allowed for a function or method body
+max-nested-blocks = 5 # (PLR1702) nested blocks within a function or method body
+max-public-methods=20 # (R0904) public methods allowed for a class
+max-returns=6 # (PLR0911) return statements for a function or method body
+max-statements=50 # (PLR0915) statements allowed for a function or method body
 
 [tool.ruff.format]
 quote-style = "single"
 indent-style = "space"
+# Consider adopting "lf" instead
```

Comment:
whats "lf"?

Reply:
Windows has a different end of line standard with an extra special character, using '\r\n\' (cr-lf pattern) while nix OS uses '\n' (lf pattern). Conforming to one standard helps to avoid strange characters showing in OSX & Linux.

BTW, on windows machine it is probably a good idea to setup global just once: `git config --global core.autocrlf true`, if you haven't already done that.

Reply:
Oh, good to know! I've had many annoying experiences with unix vs windows path standards

Reply:
> Windows has a different end of line standard with an extra special character, using '\r\n' (cr-lf pattern) while nix OS uses '\n' (lf pattern). Conforming to one standard helps to avoid strange characters showing in OSX & Linux.
> 
> BTW, on windows machine it is probably a good idea to setup global just once: `git config --global core.autocrlf true`, if you haven't already done that.

Sorry, Just to be clear. That command is for windows only.

Reply:
I'll close this one as well.